### PR TITLE
- Added Trim Functions into the ShooterWheelSubsystem

### PR DESCRIPTION
**Pull request by johnylamw**
- Added Trim Functions into the ShooterWheelSubsystem
- Added a Trim Function Test
- Added Two Instant Commands into OperatorCommandMap that needs controller assignment

### CODE REVIEW

File: src/main/java/competition/operator_interface/OperatorCommandMap.java [hunk 1]
```diff
@@ -81,6 +85,9 @@ public void setupShooterCommands(OperatorInterface operatorInterface, ShooterWhe
         operatorInterface.operatorGamepad.getifAvailable(XboxButton.RightBumper).whenPressed(speedUp);
         operatorInterface.manualOperatorGamepad.getifAvailable(XboxButton.RightBumper).whenPressed(speedUp);
         operatorInterface.manualOperatorGamepad.getifAvailable(XboxButton.LeftBumper).whenPressed(slowDown);
+
+        // assign a button to increaseTrim
```
Comment (aschokking) [inline src/main/java/competition/operator_interface/OperatorCommandMap.java]: add TODO to the comment

File: src/main/java/competition/operator_interface/OperatorCommandMap.java [hunk 2]
```diff
@@ -73,6 +73,10 @@ public void setupShooterCommands(OperatorInterface operatorInterface, ShooterWhe
 
         Command speedUp = new InstantCommand(() -> shooter.changeTargetRPM(500), shooter.getSetpointLock());
         Command slowDown = new InstantCommand(() -> shooter.changeTargetRPM(-500), shooter.getSetpointLock());
+
+        Command increaseTrim = new InstantCommand(() -> shooter.increaseTargetTrimRPM(100));
```
Comment (JohnGilb) [inline src/main/java/competition/operator_interface/OperatorCommandMap.java]: `changeTrimRPM()`

File: src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java [hunk 3]
```diff
@@ -29,6 +31,7 @@ public ShooterWheelSubsystem(CommonLibFactory factory, PropertyFactory pf, Ideal
 
         targetRpmProp = pf.createEphemeralProperty("TargetRPM", 0);
         currentRpmProp = pf.createEphemeralProperty("CurrentRPM", 0);
+        rpmTrimProp = pf.createPersistentProperty("TrimRPM", 0);
```
Comment (aschokking) [inline src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java]: change to ephemeral

Comment (JohnGilb) [inline src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java]: `ephemeral` property

**Review by aschokking — CHANGES_REQUESTED** [review]

File: src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java [hunk 4]
```diff
@@ -44,7 +47,27 @@ public void setTargetRPM(double speed) {
     }
 
     public double getTargetRPM() {
-        return targetRpmProp.get();
+        if (targetRpmProp.get() == 0) {
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java]: remember, don't compare doubles to 0. You want something closer to:

`if (Math.abs(targetRpmProp.get()) < 50)`

**Review by JohnGilb — APPROVED** [review]

**Review by aschokking — APPROVED** [review]

**Pull request merged**